src/Microsoft.AspNetCore.Mvc.Core/Filters/MiddlewareFilterAttribute.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.Core.Filters
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public class MiddlewareFilterAttribute : Attribute, IFilterFactory, IOrderedFilter
+    {
+        public MiddlewareFilterAttribute(Type type)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            ImplementationType = type;
+        }
+
+        public Type ImplementationType { get; }
```

Comment:
You copied this from `[ServiceFilter]` didn't you? BOOM :boom: caught. Seriously this should have a better name.